R/execute.R

```diff
@@ -45,30 +48,45 @@
 #' @importFrom parallel detectCores
 #' @importFrom parallel mcmapply
 #' @importFrom berryFunctions tryStack
+#' @importFrom utils installed.packages
 #' 
 #' @name execute
 #' @export
-execute_multiverse <- function(multiverse, cores = getOption("mc.cores", 1L)) {
+execute_multiverse <- function(multiverse, cores = getOption("mc.cores", 1L), progress = FALSE) {
   m_obj <- attr(multiverse, "multiverse")
   m_diction = attr(multiverse, "multiverse")$multiverse_diction
   
   .level = min(m_obj$unchanged_until, m_obj$exec_all_until)
   .to_exec = tail(seq_len(m_diction$size()), n = m_diction$size() - .level)
   
   .m_list <- m_diction$as_list()[.to_exec] # list of unexecuted (vertical) steps in the multiverse
-  .res <- lapply(.m_list, exec_all, cores = cores) # we execute each step in sequence from top to bottom
+  
+  if (progress && !("pbmcapply" %in% rownames(installed.packages()))) {
```

Comment:
Minor point: the typical form of `"pbmcapply" %in% rownames(installed.packages())` --- a check for package availability --- is something like `requireNamespace("pbmcapply", quietly = TRUE)`. The latter will generally be faster than using `installed.packages()` for the check.

Reply:
Yeah, I figured this may not be optimal. I was thinking of addressing it by making changes to the dev branch, before merging to master

Reply:
Thanks for the tip!